src/main/permissionsManager.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {
+    PermissionRequestHandlerHandlerDetails,
+    WebContents,
+    app,
+    dialog,
+    ipcMain,
+} from 'electron';
+
+import {UPDATE_PATHS} from 'common/communication';
+import JsonFileManager from 'common/JsonFileManager';
+import {Logger} from 'common/log';
+import {t} from 'common/utils/util';
+import {isTrustedURL, parseURL} from 'common/utils/url';
+
+import {permissionsJson} from 'main/constants';
+import {localizeMessage} from 'main/i18nManager';
+import ViewManager from 'main/views/viewManager';
+import CallsWidgetWindow from 'main/windows/callsWidgetWindow';
+import MainWindow from 'main/windows/mainWindow';
+
+const log = new Logger('PermissionsManager');
+
+// supported permission types
+const supportedPermissionTypes = [
+    'media',
+    'geolocation',
+    'notifications',
+    'fullscreen',
+    'openExternal',
+    'clipboard-sanitized-write',
+];
+
+// permissions that require a dialog
+const authorizablePermissionTypes = [
+    'media',
+    'geolocation',
+    'notifications',
+];
+
+type Permissions = {
+    [origin: string]: {
+        [permission: string]: {
+            allowed: boolean;
+            alwaysDeny?: boolean;
+        };
+    };
+};
+
+export class PermissionsManager extends JsonFileManager<Permissions> {
+    handlePermissionRequest = async (
+        webContents: WebContents,
+        permission: string,
+        callback: (granted: boolean) => void,
+        details: PermissionRequestHandlerHandlerDetails,
+    ) => {
+        callback(await this.doPermissionRequest(
+            webContents.id,
+            permission,
+            details.securityOrigin ?? details.requestingUrl,
+        ));
+    }
+
+    doPermissionRequest = async (
+        webContentsId: number,
+        permission: string,
+        requestingURL: string,
+    ) => {
+        log.debug('doPermissionRequest', requestingURL, permission);
+
+        // is the requested permission type supported?
+        if (!supportedPermissionTypes.includes(permission)) {
```

Comment:
You may want to consider adding an explicit type for `permission`. If you do it as a union type or string enum, you shouldn't need to confirm if the permission is supported or not

Reply:
I tried working with that, the problem is that there is a set of constants Electron allowed, a set we allow, and a set we allow but also require user interaction for. It really wasn't providing any extra value, and we still have to check that the permission coming from the handler is one of our supported ones.